### DISCUSSION TITLE
Questions in LSTM tutorial

### DISCUSSION THREAD

**Discussion by NeroBlackstone** [LuxDL/Lux.jl#695] [Q&A] [answered]
In section [Using the @compact API](https://lux.csail.mit.edu/stable/tutorials/beginner/3_SimpleRNN#Using-the-@compact-API):

``` julia
x_init, x_rest = Iterators.peel(Lux._eachslice(x, Val(2)))
```

I don't quite understand this line of code. What does Val(2) do here?

**Comment by avik-pal** [comment, marked as answer]
It is same as the `dims` kwarg in `eachslice`, but passing in `Val(2)` for type-stability
